### ISSUE TITLE
Calling xr.open_dataset in open_sat_data on a path with a wildcard in it?

### ISSUE THREAD

**Issue by JackKelly** [openclimatefix/nowcasting_dataset#435]
The logs are a bit ambiguous but I think what's going on is that we're calling xr.open_dataset in open_sat_data on a path with a wildcard in it?

```
concurrent.futures.process._RemoteTraceback: 
"""
2021-11-17 18:13:00,506 DEBUG processID=1556849 Opening satellite data: /mnt/storage_ssd_8tb/data/ocf/solar_pv_nowcasting/nowcasting_dataset_pipeline/satellite/EUMETSAT/SEVIRI_RSS/zarr/v2/eumetsat_zarr_* | /home/jack/dev/ocf/nowcasting_dataset/nowcasting_dataset/data_sources/satellite/satellite_data_source.py#L239


....

2021-11-17 18:16:17,323 ERROR processID=1555649 Worker process satellite raised exception!
'.zmetadata' | /home/jack/dev/ocf/nowcasting_dataset/nowcasting_dataset/manager.py#L457
NoneType: None
ERROR:nowcasting_dataset.manager:Worker process satellite raised exception!
'.zmetadata'
NoneType: None

...

Traceback (most recent call last):
  File "/home/jack/miniconda3/envs/nowcasting_dataset/lib/python3.9/concurrent/futures/process.py", line 243, in _process_worker
    r = call_item.fn(*call_item.args, **call_item.kwargs)
  File "/home/jack/dev/ocf/nowcasting_dataset/nowcasting_dataset/data_sources/data_source.py", line 178, in create_batches
    self.open()
  File "/home/jack/dev/ocf/nowcasting_dataset/nowcasting_dataset/data_sources/satellite/satellite_data_source.py", line 48, in open
    self._data = self._open_data()
  File "/home/jack/dev/ocf/nowcasting_dataset/nowcasting_dataset/data_sources/satellite/satellite_data_source.py", line 52, in _open_data
    return open_sat_data(zarr_path=self.zarr_path, consolidated=self.consolidated)
  File "/home/jack/dev/ocf/nowcasting_dataset/nowcasting_dataset/data_sources/satellite/satellite_data_source.py", line 248, in open_sat_data
    dataset = xr.open_dataset(
  File "/home/jack/miniconda3/envs/nowcasting_dataset/lib/python3.9/site-packages/xarray/backends/api.py", line 495, in open_dataset
    backend_ds = backend.open_dataset(
  File "/home/jack/miniconda3/envs/nowcasting_dataset/lib/python3.9/site-packages/xarray/backends/zarr.py", line 824, in open_dataset
    store = ZarrStore.open_group(
  File "/home/jack/miniconda3/envs/nowcasting_dataset/lib/python3.9/site-packages/xarray/backends/zarr.py", line 384, in open_group
    zarr_group = zarr.open_consolidated(store, **open_kwargs)
  File "/home/jack/miniconda3/envs/nowcasting_dataset/lib/python3.9/site-packages/zarr/convenience.py", line 1183, in open_consolidated
    meta_store = ConsolidatedMetadataStore(store, metadata_key=metadata_key)
  File "/home/jack/miniconda3/envs/nowcasting_dataset/lib/python3.9/site-packages/zarr/storage.py", line 2590, in __init__
    meta = json_loads(store[metadata_key])
  File "/home/jack/miniconda3/envs/nowcasting_dataset/lib/python3.9/site-packages/zarr/storage.py", line 847, in __getitem__
    raise KeyError(key)
KeyError: '.zmetadata'
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/jack/dev/ocf/nowcasting_dataset/scripts/prepare_ml_data.py", line 79, in <module>
    main()
  File "/home/jack/miniconda3/envs/nowcasting_dataset/lib/python3.9/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/jack/miniconda3/envs/nowcasting_dataset/lib/python3.9/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/jack/miniconda3/envs/nowcasting_dataset/lib/python3.9/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/jack/miniconda3/envs/nowcasting_dataset/lib/python3.9/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/jack/dev/ocf/nowcasting_dataset/nowcasting_dataset/utils.py", line 155, in inner_func
    return func(*args, **kwargs)
  File "/home/jack/dev/ocf/nowcasting_dataset/scripts/prepare_ml_data.py", line 72, in main
    manager.create_batches(overwrite_batches)
  File "/home/jack/dev/ocf/nowcasting_dataset/nowcasting_dataset/manager.py", line 460, in create_batches
    raise exception
KeyError: '.zmetadata'

```

**Comment by JackKelly** [comment]
I'll look into this now :slightly_smiling_face:

**Comment by peterdudfield** [comment]
https://github.com/openclimatefix/nowcasting_dataset/blob/main/nowcasting_dataset/data_sources/satellite/satellite_data_source.py#L249

Does that bit of code help?

**Comment by JackKelly** [comment]
I think this fixes is: PR #436

**Comment by jacobbieker** [comment]
Adding `()` to the end of `Path(zarr_path).exists:` fixes it
